docs/migrations/chain-upgrade-guide-040.md

```diff
@@ -0,0 +1,163 @@
+<!--
+order: 2
+-->
+
+# Chain Upgrade Guide to v0.40
+
+This document explains how to perform a chain upgrade from v0.39 to v0.40. {synopsis}
+
+## Risks
+
+As a validator, performing the upgrade procedure on your consensus nodes carries a heightened risk of double-signing and
+being slashed. The most important piece of this procedure is verifying your software version and genesis file hash before
+starting your validator and signing.
```

Comment:
Can you explain when a validator can double sign? How genesis file hash validation will prevent that?

Reply:
Does the line below explain it well enough?

"The riskiest thing a validator can do is to discover that they made a mistake and repeat the upgrade procedure again during
the network startup."

Reply:
No, because if I do upgrade I don't expect to sign anything. Why upgrade involves signing? I guess we will vote for some blocks in the meantime, and if we restart quickly (during the same block time) we can do a double vote, right? If yes, then let's describe it.

Reply:
Ah i think i understood differently. For me you upgrade and restart the node at N, then vote a block at N+1. But then you figured you made a mistake in the migration (so wrong genesis hash), so you restart again at N, then vote again at N+1, leading to double vote.

@anilCSE could you clarify what you meant here?

Reply:
If you made a mistake, then:

* if you are a block producer then you create a not valid block (wrong genesis hash) so nobody will vote for it
* if you receive a block and the chain doesn't match your chain (because you have different genesis) then you will not vote for it.

Reply:
Updated.